internal/servers/controller/handlers/authmethods/authmethod_service.go

```diff
@@ -239,6 +239,8 @@ func (s Service) DeleteAuthMethod(ctx context.Context, req *pbs.DeleteAuthMethod
 }
 
 // Authenticate implements the interface pbs.AuthenticationServiceServer.
+//
+// Deprecated in favor of AuthenticateLogin
 func (s Service) Authenticate(ctx context.Context, req *pbs.AuthenticateRequest) (*pbs.AuthenticateResponse, error) {
```

Comment:
Instead of duplicating the logic consider replacing the body with:
```
r, err := s.AuthenticateLogin(ctx, &pbs.AuthenticateLoginRequest{req.blabla})
// checks, etc...
return &pbs.AuthenticateResponse{r.blablabla}, err
```
Or since the AuthenticateRequest/Response is JSON equivalent to AuthenticateLoginRequest/Response consider replacing the former with the latter in the proto (and thus here as well so it can just be a passthrough)

Reply:
If you re-use a message in the proto it won't compile, that's why there are different messages to begin with.

Reply:
I'm going to leave as-is as we're going to be removing the original function in a couple of releases. Not worth combining them to reduce duplication and then having to pull it apart again.